Enable basic tubes/process test on Windows

peace-maker opened the pull request:
Many tests are Linux specific using qemu or /proc etc. So only test the basic process interaction for now.

peace-maker committed:
Enable basic tubes/process test on Windows

Many tests are Linux specific using qemu or /proc etc. So only test the basic process interaction for now.

Arusekk commented on pwnlib/tubes/process.py:
```diff
@@ -135,19 +135,22 @@ class process(tube):
-        >>> p.recvline()
-        b'Hello world\n'
+        >>> p.recvline() # doctest: +ELLIPSIS
+        b'Hello world...\n'
```
I get it, it is great to test things work, but does this still have the value of being obvious documentation for the function?

peace-maker replied:
Yeah, it's annoying. This goes away after `recvline` doesn't include the newline anymore in #2588. We could just add `drop=True` explicitly too for this test?

peace-maker committed:
Cleanup docs without using ellipsis for Windows output

peace-maker committed:
Merge branch 'dev' into windows_process_test

peace-maker committed:
Set newline to b'\r\n' for local Windows processes

But only change it automatically if `context.newline` wasn't set manually.

peace-maker committed:
Disable process.terminate() test on Windows

Windows doesn't support sending SIGTERM to other processes.

https://docs.python.org/3/library/subprocess.html#subprocess.Popen.send_signal

peace-maker committed:
libcdb CI cache: add another dns server

[emerg] 1#1: host not found in upstream "debuginfod.elfutils.org" in /etc/nginx/nginx.conf:26

The cache should still start despite not being able to resolve some domain.
